datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1399,6 +1399,18 @@ impl<S: SimplifyInfo> TreeNodeRewriter for Simplifier<'_, S> {
             // Rules for Case
             //
 
+            // CASE WHEN true THEN A ... END --> A
+            Expr::Case(Case {
+                expr: None,
+                mut when_then_expr,
+                else_expr: _,
+            }) if !when_then_expr.is_empty()
```

Comment:
Reading this code again, it seems like we could remove all WHEN THEN clauses with `true`, not just the first

However, this is an improvement over what is on main so it would be a good follow on pR

Reply:
- Filed https://github.com/apache/datafusion/issues/17589 to track